src/strategies/SiloFarmStrategy.sol

```diff
@@ -0,0 +1,275 @@
+        if (addresses.length != 2 || nums.length != 1 || ticks.length != 0) {
+            revert IControllable.IncorrectInitParams();
+        }
+
+        IFactory.Farm memory farm = _getFarm(addresses[0], nums[0]);
+        if (farm.addresses.length != 2 || farm.nums.length != 0 || farm.ticks.length != 0) {
+            revert IFarmingStrategy.BadFarm();
+        }
+
+        address[] memory siloAssets = new address[](1);
+        ISilo siloVault = ISilo(farm.addresses[1]);
+        siloAssets[0] = siloVault.asset();
+        __StrategyBase_init(addresses[0], StrategyIdLib.SILO_FARM, addresses[1], siloAssets, address(0), 0);
+        __FarmingStrategyBase_init(addresses[0], nums[0]);
+        IERC20(siloAssets[0]).forceApprove(farm.addresses[1], type(uint).max);
+        IERC20(farm.addresses[1]).forceApprove(farm.addresses[0], type(uint).max);
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                       VIEW FUNCTIONS                       */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /// @inheritdoc IStrategy
+    function strategyLogicId() public pure override returns (string memory) {
+        return StrategyIdLib.SILO_FARM;
+    }
+
+    /// @inheritdoc IStrategy
+    function extra() external pure returns (bytes32) {
+        return CommonLib.bytesToBytes32(abi.encodePacked(bytes3(0x00d395), bytes3(0x000000)));
+    }
+
+    /// @inheritdoc IStrategy
+    function description() external view returns (string memory) {
+        IFactory.Farm memory farm = _getFarm();
+        return _genDesc(farm);
+    }
+
+    /// @inheritdoc IStrategy
+    function getAssetsProportions() external pure returns (uint[] memory proportions) {
+        proportions = new uint[](1);
+        proportions[0] = 1e18;
+    }
+
+    /// @inheritdoc IStrategy
+    function getRevenue() external view returns (address[] memory __assets, uint[] memory amounts) {}
+
+    /// @inheritdoc IStrategy
+    function getSpecificName() external pure override returns (string memory, bool) {
+        return ("", false);
+    }
+
+    /// @inheritdoc IStrategy
+    function supportedVaultTypes() external pure override returns (string[] memory types) {
+        types = new string[](1);
+        types[0] = VaultTypeLib.COMPOUNDING;
+    }
+
+    /// @inheritdoc IStrategy
+    function initVariants(address platform_)
+        public
+        view
+        returns (string[] memory variants, address[] memory addresses, uint[] memory nums, int24[] memory ticks)
+    {
+        addresses = new address[](0);
+        ticks = new int24[](0);
+        IFactory.Farm[] memory farms = IFactory(IPlatform(platform_).factory()).farms();
+        uint len = farms.length;
+        variants = new string[](1);
+        nums = new uint[](1);
+        //slither-disable-next-line uninitialized-local
+        uint total;
```

Comment:
rename to `_total` to mitigate slither's shadowing
because `total` is `IStrategy.total()`